File: unix/boot/spp/rpp/ratlibr/stcopy.r
include	defs

# stcopy - copy string from in (i) to out (j), updating j, excluding EOS

   subroutine stcopy (in, i, out, j)
   character in (ARB), out (ARB)
   integer i, j

   integer k

   for (k = i; in (k) != EOS; k = k + 1) {
      out (j) = in (k)
      j = j + 1
      }
   out(j) = EOS
   return
   end

File: unix/boot/spp/rpp/ratlibf/stcopy.f
      SUBROUTINE STCOPY (IN, I, OUT, J)
      INTEGER IN (100), OUT (100)
      INTEGER I, J
      INTEGER K
      K = I
23000 IF (.NOT.(IN (K) .NE. -2))GOTO 23002
      OUT (J) = IN (K)
      J = J + 1
23001 K = K + 1
      GOTO 23000
23002 CONTINUE
      OUT(J) = -2
      RETURN
      END
